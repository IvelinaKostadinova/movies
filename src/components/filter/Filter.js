import React from 'react';
import PropTypes from 'prop-types';

import './Filter.scss';

const Filter = (props) => {
  const handleClick = () => {
    props.handleClick(props.text);
  };

  return (
    <button
      className={`${props.active ? 'filter--active' : 'filter'}`}
      type="submit"
      onClick={handleClick}
    >
      {props.text}
    </button>
  );
};

Filter.propTypes = {
  text: PropTypes.string,
  handleClick: PropTypes.func,
  active: PropTypes.bool,
};

export default Filter;
