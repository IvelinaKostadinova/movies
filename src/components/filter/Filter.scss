@import '../../styles/variables';

.filter {
    background-color: transparent;
    color: $white;
    border-color: transparent;
    cursor: pointer;
    margin: 2rem;
    font-size: 1.8rem;
    height: 4.4rem;

    &--active {
      background-color: transparent;
      color: $white;
      border-color: transparent;
      cursor: pointer;
      margin: 2rem;
      font-size: 1.8rem;
      height: 4.4rem;
      border-bottom: 0.3rem solid $pink;

      &:focus, &:active, &:visited {
        outline: none;
      }

      &:hover {
        color: $pink;
      }
    }

    &:focus, &:active, &:visited {
      outline: none;
    }

    &:hover {
      color: $pink;
    }
}