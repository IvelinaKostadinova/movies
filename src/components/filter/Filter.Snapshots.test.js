import React from 'react';
import Filter from './Filter';
import renderer from 'react-test-renderer';

it('sets genre filter to COMEDY', () => {
  const tree = renderer.create(
    <Filter text={'COMEDY'} handleClick={jest.fn()}></Filter>
  );

  expect(tree).toMatchSnapshot();
});
