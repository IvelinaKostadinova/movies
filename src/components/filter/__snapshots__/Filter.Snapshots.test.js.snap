// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets genre filter 1`] = `
<button
  className="filter"
  onClick={[Function]}
  type="submit"
>
  COMEDY
</button>
`;

exports[`sets genre filter to COMEDY 1`] = `
<button
  className="filter"
  onClick={[Function]}
  type="submit"
>
  COMEDY
</button>
`;
